perf/src/sigverify.rs

```diff
@@ -496,6 +503,46 @@ impl Deduper {
     }
 }
 
+//inplace shrink a batch of packets
+pub fn shrink_batches(batches: &mut Vec<PacketBatch>) -> usize {
+    let mut keep_batch_index = 0;
+    let mut keep_packet_index = 0;
+    let mut shrunken_length = 0;
+    for batch_index in 0..batches.len() {
+        for packet_index in 0..batches[batch_index].packets.len() {
+            if batches[batch_index].packets[packet_index].meta.discard() {
+                continue;
+            }
+            'concentrate_kept: while keep_batch_index < batch_index {
+                while keep_packet_index < batches[keep_batch_index].packets.len() {
+                    let this_keep_packet_index = keep_packet_index;
+                    solana_sdk::saturating_add_assign!(keep_packet_index, 1);
```

Comment:
this macro is as long as std apis, but now I have to think about wtf it does

Reply:
It was way shorter before I hit the `info!` symbol conflict between `solana_sdk` via `solana_program` and `log` :face_with_head_bandage: 

This is mainly a marker for code to be updated with `std::num::Saturating` once it stablizes.  The naming is a portmanteau of existing names from `std`, `saturating_add` and `add_assign`, should be fairly straight forward to deduce the behavior